src/k2/transport/Payload.cpp

```diff
@@ -391,6 +391,34 @@ Payload Payload::share() {
     return shared;
 }
 
+
+Payload Payload::share(size_t nbytes) {
```

Comment:
This code is nearly identical to share() but it is different. share() shares the entire payload, regarding of current position, so share(nbytes) doesn't quite make sense as a method name and intended functionality.
I would rework the api a bit to reflect that:
Payload shareAll() ; // share entire payload
Payload shareRegion(PayloadPosition start, size_t nbytes); // share a particular region from the payload.

the new `shareRegion` method would be a slightly modified version of the existing `share()` method.

then shareAll() implementation is just:
```c++
return shareRegion(PayloadPosition(0,0,0), getSize());
```

Reply:
rewrite the share to shareAll and shareRegion